src/java.base/share/classes/java/lang/foreign/SegmentAllocator.java

```diff
@@ -90,209 +90,283 @@ public interface SegmentAllocator {
+     * initialized with the provided float {@code value} using the byte order of the provided
+     * {@linkplain ValueLayout#order() layout.order()}}
+     *
+     * @implSpec The default implementation of this method first calls {@code this.allocate(layout)} to
+     * allocate a segment after which the provided {@code value} is written into the segment using the
+     * byte ordering specified by the provided {@code layout}.
+     *
      * @param layout the layout of the block of memory to be allocated.
-     * @param value the value to be set on the newly allocated memory block.
-     * @return a segment for the newly allocated memory block.
+     * @param value  the value to be set in the newly allocated memory segment.
      */
     default MemorySegment allocate(ValueLayout.OfFloat layout, float value) {
         Objects.requireNonNull(layout);
         VarHandle handle = layout.varHandle();
-        MemorySegment addr = allocate(layout);
-        handle.set(addr, value);
-        return addr;
+        MemorySegment seg = allocate(layout);
+        handle.set(seg, value);
+        return seg;
     }
 
     /**
-     * Allocates a memory segment with the given layout and initializes it with the given long value.
-     * @implSpec the default implementation for this method calls {@code this.allocate(layout)}.
+     * {@return a newly allocated memory segment with a {@linkplain MemorySegment#byteSize() byteSize()} of 8
+     * initialized with the provided long {@code value} using the byte order of the provided
+     * {@linkplain ValueLayout#order() layout.order()}}
+     *
+     * @implSpec The default implementation of this method first calls {@code this.allocate(layout)} to
+     * allocate a segment after which the provided {@code value} is written into the segment using the
+     * byte ordering specified by the provided {@code layout}.
+     *
      * @param layout the layout of the block of memory to be allocated.
-     * @param value the value to be set on the newly allocated memory block.
-     * @return a segment for the newly allocated memory block.
+     * @param value  the value to be set in the newly allocated memory segment.
      */
     default MemorySegment allocate(ValueLayout.OfLong layout, long value) {
         Objects.requireNonNull(layout);
         VarHandle handle = layout.varHandle();
-        MemorySegment addr = allocate(layout);
-        handle.set(addr, value);
-        return addr;
+        MemorySegment seg = allocate(layout);
+        handle.set(seg, value);
+        return seg;
     }
 
     /**
-     * Allocates a memory segment with the given layout and initializes it with the given double value.
-     * @implSpec the default implementation for this method calls {@code this.allocate(layout)}.
+     * {@return a newly allocated memory segment with a {@linkplain MemorySegment#byteSize() byteSize()} of 4
```

Comment:
Size should be 8 here - but in general I think I'd rather leave it more general and not to specify sizes. The reader should be able to follow the javadoc, and get that we end up calling `allocate(long, long)` with certain parameters (which are derived from the layout). I don't think being extra precise here helps - and also opens up to issues: note that the alignment of double/long layout constants is platform-specific, which I don't think you want to get into here?